tools/analyzer_plugin/lib/src/diagnostic/missing_required_prop.dart

```diff
@@ -53,62 +54,121 @@ render() {
 // </editor-fold>
 
 class MissingRequiredPropDiagnostic extends ComponentUsageDiagnosticContributor {
```

Comment:
Since this diagnostic always existed but was only partially implemented, was always disabled, and was never fully reviewed, this full file should be reviewed as if it were newly-added, as opposed to reviewing just the diff